compress/lib.h

```diff
@@ -30,6 +30,18 @@
  * | compress/lz4.c      | @subpage compress_lz4  |
  * | compress/zlib.c     | @subpage compress_zlib |
  * | compress/zstd.c     | @subpage compress_zstd |
+ *
+ * Default usage with Compression Level set to 1:
+ * open() -> compress() -> close()
+ * open() -> decompress() -> close()
+ *
+ * Usage with Compression Level set to N:
+ * open() -> set_level(N) -> compress() -> close()
+ * open() -> decompress() -> close()
+ *
+ * Usage with Zstd Dictionary Compression and Level:
+ * open() -> set_level(N) -> set_dictionary() -> compress() -> close()
+ * open() -> set_dictionary() -> decompress() -> close()
```

Comment:
These comments are misleading, as they suggest a single `compress` for a single `open`/`close` pairs. In reality, at least for IMAP, you often do one `open`, followed by a number of `compress`, followed by one `close`. This is why I suggest to move `set_level` and `set_dictionary` inside `open` instead of inside `store`.

Reply:
I asked @flatcap before, he wanted set_level() and set_dictionary() instead of open(level, dictionary).